### PR TITLE
cmake build update

### PR DESCRIPTION
**Pull request by senior7515**
update build for seastar/cmake with cooking.sh

### CODE REVIEW

**Comment by noahdesu** [comment]
same question

**Comment by noahdesu** [comment]
basically it looks like importing the ingredients etc from seastar?

**Comment by senior7515** [comment]
@noahdesu - at first glance it looks like that. 

The compile flags are different and override those in seastar for the 2 overrides (which is just the beginning - i want to statically link the rest of the libs with the default `wellknown` cmake file). The others zlib - we need and lz4 we need, fmt we need independently. This is how cooking.sh works tho. (more on that later)

The executive summary:

- Provide *one* way of supplying default deps - via building them.
  - This doesn't mean that the user is not allowed to supply their own. Just means that the default will just download it and use it. 
  - so say ceph wants to use it, they will not use any recipe stuff. they will simply do `find_package(<smf deps>)` before calling `link_target_... ( <foo> smf::smf  ... )` 
  - i hope this is just step 1 of many more fixes to come. 
 
- Simplify the build symbols for SMF. no more SMF_DEBUG (change to NDEBUG), removed the dependency on bytell hash map - measured for normal requests and std::unordered map does not too bad. The request dispatch is not on the hotpath. And a few minor things like that to the build.

Let me know if this makes sense.

File: recipe/dpdk_config [hunk 1]
```diff
@@ -0,0 +1,23 @@
+CONFIG_RTE_LIBRTE_PMD_BOND=n
```
Comment (senior7515) [inline recipe/dpdk_config]: I think i can remove this.

File: recipe/dpdk_configure.cmake [hunk 2]
```diff
@@ -0,0 +1,40 @@
+#
```
Comment (senior7515) [inline recipe/dpdk_configure.cmake]: and this.

**Comment by senior7515** [comment]
@noahdesu  - so if you look at the actual files, i don't think it is like you said. The latest commit has a more cleaned up version of the code. Let me know if you have any other thoughts.

**Comment by senior7515** [comment]
@noahdesu other source of `changes` which are empty is that cooking.sh expects the cmake folder to be named `cmake` not `CMake` - jesse is fixing upstream. for now I don't mind the rename since it enables me to keep writing code.

**Comment by senior7515** [comment]
This is missing travis test passes. *sigh* ... testing local.sh and seeing where it hangs.

**Pull request merged**